Automattic/woocommerce-payments
# Fix fraud prevention token not showing up on checkout pages created via the site editor

Fixes #8141

#### Changes proposed in this Pull Request



This PR changes the way the fraud prevention token is delivered to the browser, and retrieved by the payment methods to be sent to the checkout endpoints. The token will now be present as a global JS variable `window.wcpayFraudPreventionToken` when the user is viewing the cart page, or the checkout page, when the fraud mitigations are activated for the merchant, instead of being appended as a value in a hidden input. 







#### Testing instructions





All the possible checkout types that use our server needs to be tested to see if the payment is still working as intended, with the fraud mitigations disabled, and enabled.

- Regular checkout using test card
- Regular checkout using saved card
- Regular checkout using one UPE method
- ApplePay
- GooglePay
- WooPay

**A way to test those in a single installation:**

* Create a JN site using the "[Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/8141-fraud-prevention-token-site-editor-problem)" link in the below comment of "botwoo".
* On the new JN site complete the WC, WooPayments and Stripe onboarding.
* Install the Storefront theme.
* Go to **Appearance > Storefront** page and click the button with both checkboxes selected. This will prepare the demo site with adding example products.
* On the Customizer which opens after it, click Publish.
* Return to your store and check if the products are there, and the front page shows the categories and products.
* Note the blog ID attached to your new blog on the WCPay Dev Tools page.
* Go to https://mc.a8c.com/woocommerce-payments/index.php and search your new blog ID.
* On the dashboard for your account, enable the card testing mitigations with setting the `card_testing_prevention_enabled` account meta to `1`.
* Refresh your account cache using the "clear cache" link on WCPay Dev Tools. 
* The site should be ready to test the above payment methods. If you don't see the payment form, check if WooPayments is enabled on **Payments > Settings**.

> [!IMPORTANT]
> Repeat the tests after completing all tests with switching to the Twenty Twenty Four theme. This will test it with a Site Editor enabled theme, which was the main problem in the issue.

-------------------


**Post merge**
